apps/src/templates/instructions/TopInstructionsCSF.jsx

```diff
@@ -154,28 +225,28 @@ var TopInstructions = React.createClass({
 
     const maxNeededHeight = this.adjustMaxNeededHeight();
 
-    // Update right col width now that we know how much space it needs. One thing
-    // to note is that if we end up resizing our column significantly, it can
-    // result in our maxNeededHeight being inaccurate. This isn't that big a deal
-    // except that it means when we set instructionsRenderedHeight below, it might
-    // not be as large as we want.
-    this.setState({
-      rightColWidth: $(ReactDOM.findDOMNode(this.refs.collapser)).outerWidth(true)
-    });
-
     // Initially set to 300. This might be adjusted when InstructionsWithWorkspace
     // adjusts max height.
     this.props.setInstructionsRenderedHeight(Math.min(maxNeededHeight, 300));
   },
 
   /**
-   * Height can get below min height iff we resize the window to be super small.
-   * If we then resize it to be larger again, we want to increase height.
+   * When collapsed, height can change when we get additional feedback
+   * or the hint prompt. In that case, we want to always resize.
+   * When in resize mode, height can get below min height iff we resize
+   * the window to be super small.  If we then resize it to be larger
+   * again, we want to increase height.
    */
   componentWillReceiveProps(nextProps) {
     const minHeight = this.getMinHeight(nextProps.collapsed);
-    if (nextProps.height < minHeight && nextProps.height < nextProps.maxHeight) {
-      this.props.setInstructionsRenderedHeight(Math.min(nextProps.maxHeight, minHeight));
+    const newHeight = Math.min(nextProps.maxHeight, minHeight);
+
+    const shouldUpdateHeight = (nextProps.collapsed)
+        ? newHeight !== this.props.height
+        : nextProps.height < minHeight && nextProps.height < nextProps.maxHeight;
```

Comment:
hah, I totally couldn't remember which we preferred, so I checked and [totally misinterpreted this rule](https://github.com/code-dot-org/code-dot-org/blob/staging/STYLEGUIDE.md#js-operator-continuations). D'oh.